app/src/main/java/fr/neamar/kiss/ApplicationLifecycleHandler.java

```diff
@@ -0,0 +1,78 @@
+package fr.neamar.kiss;
+
+import android.app.Activity;
+import android.app.Application;
+import android.content.ComponentCallbacks2;
+import android.content.ComponentName;
+import android.content.res.Configuration;
+import android.os.Bundle;
+
+import fr.neamar.kiss.cache.MemoryCacheHelper;
+import fr.neamar.kiss.pojo.AppPojo;
+import fr.neamar.kiss.pojo.Pojo;
+
+public class ApplicationLifecycleHandler implements Application.ActivityLifecycleCallbacks, ComponentCallbacks2 {
+
+    //private static final String TAG = ApplicationLifecycleHandler.class.getSimpleName();
+    private static boolean isInBackground = false;
+
+    @Override
+    public void onActivityCreated(Activity activity, Bundle bundle) {
+    }
+
+    @Override
+    public void onActivityStarted(Activity activity) {
+    }
+
+    @Override
+    public void onActivityResumed(Activity activity) {
+
+        if (isInBackground) {
+            //Log.d(TAG, "app went to foreground");
+            isInBackground = false;
+
+            for (Pojo pojo : KissApplication.getApplication(activity).getDataHandler().getApplications()) {
+                if (!(pojo instanceof AppPojo))
```

Comment:
[Generally using `instanceof` is a bad idea](http://www.javapractices.com/topic/TopicAction.do?Id=31). You probably should use polymorphism here. Something like ```if(pojo.isPreloadable()) {``` maybe?

Reply:
This would bloat the base pojo object with unnecessary methods especially when I am 100% sure that the pojo I will get from **KissApplication.getApplication(activity).getDataHandler().getApplications()** is an AppPojo.
I was more so avoiding the cast without check warning here.